src/index.js

```diff
@@ -654,6 +657,10 @@ export function setQueryData(
 
   const query = queries.find(d => d.queryHash === queryHash)
 
+  if (!query) {
+    return
```

Comment:
Here we can probably create the query object instead.

Reply:
Yeah, we could create the query here. Do you have a use-case for that? Honestly, it could be the same as (or share logic from) `prefetchQuery`

Reply:
No I don't have a use case for it. However I worked around this bug by calling prefetchQuery prior to running setQueryData. Would you prefer doing the same thing as prefetchQuery?

Reply:
I tried using the same logic as prefetchQuery, but I couldn't get it to work. The tests seems to have some race-conditions and I feel that it's generally become very unpredictable how the different code paths reacts to a query without a query function provided.

I can push the code if you want to have a look.

Reply:
Yeah I’ll take a look